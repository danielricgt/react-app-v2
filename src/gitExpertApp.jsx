import { useState } from "react"
import { AddCategory } from "./components/addCategory";

export const GitExpertApp = () => {

    const [categories, setCategories] = useState(['Dragon Ball', 'One Piece', 'Naruto']);
    // console.log(categories);

    const onAddCategory = () => {
        //  setCategories([...categories, 'HunterXHunter']);
        setCategories(cat => [...cat, 'HunterXHunter']);
    }
    return (
        <>
            {/* TITULO */}
            {/* INPUT */}
            <AddCategory/>
            {/* GIFS ITEM LIST  */}
            <button onClick={onAddCategory}>Agregar</button>
            <h1>Hola Mundo</h1>
            <h2>Git Expert App</h2>
            <ol>
                {categories.map(category => {
                    return <li key={category}>{category}</li>
                })}
            </ol>
            {/* GIF ITEM  */}
        </>
    )
}
