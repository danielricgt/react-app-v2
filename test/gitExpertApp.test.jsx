import { render, screen } from "@testing-library/react";
import{GitExpertApp} from '../src/gitExpertApp'


jest.mock("../src/config/apiKey", () => ({
    __esModule: true,
    default: "mocked_api_key",
  }));

describe('git expert app test', () => { 

    test('should render the app and match the snapshop', () => {
        const {container} = render(<GitExpertApp/>);
        expect(container).toMatchSnapshot();
    });
    test('should not return new categorie if it already exist', () => {
        const onAddCategory = jest.fn();
        const newCategory = 'Dragon Ball';
        const categories = ['One Punch', 'Death Note', 'Dragon Ball'];
        render(<GitExpertApp/>);
        expect(onAddCategory).not.toHaveBeenCalled();

        

    });
    

    
})