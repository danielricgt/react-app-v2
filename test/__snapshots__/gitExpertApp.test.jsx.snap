// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`git expert app test should render the app and match the snapshop 1`] = `
<div>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar Gifs"
      type="text"
      value=""
    />
  </form>
  <h1>
    Hola Mundo
  </h1>
  <h2>
    Git Expert App
  </h2>
</div>
`;
